### DISCUSSION TITLE
Combine a style with the existing style?

### DISCUSSION THREAD

**Discussion by dphaine** [DeepSpace2/StyleFrame#146] [Ideas]
As I understand it, it's possible to 
- set the global default for all cells
- completely overwrite the default with a different style
- combine a new style with the global default style

But what doesn't seem directly possible is to:
- combine a new style with whatever a cell's current style is

For example: I have certain columns left-aligned and some right-aligned.  Certain rows I want to boldface, to emphasize.  And now, certain columns I want to colorize so they appear grouped together.

Unless I'm missing something, this is pretty tricky to code around; it creates a combinatorics of formatting styles.

What do you think of an option when applying a style to "combine with existing"?  This way, complimentary formatting rules could be applied without each having to know about the others.

**Comment by DeepSpace2** [comment]
Hi,
so apparently this was what the `overwrite_default_style` argument should have been doing all-along, there was just a small oversight when we added support for reading Excel files with their style...

I'm wondering if we should "fix" this or introduce a new boolean argument (`combine_with_current_style` maybe?) to all of the styling methods (`apply_style_by_indexes`, `apply_column_style` and `apply_headers_style`).

**Comment by dphaine** [comment]
Hmm perhaps some people are depending on the current behavior.  

Introducing a new `combine_with_current_style` would save code and processing time!
